compiler/luci/pass/src/QuantizeWeightsWithGPTQPass.test.cpp

```diff
@@ -0,0 +1,209 @@
+  // Since output type is FLOAT32, an exception is expected
+  EXPECT_THROW(pass.run(&_g), std::runtime_error);
+}
+
+// Negative test: Provide an empty hessian map
+TEST_F(QuantizeWeightsWithGPTQPassTest, run_with_empty_hessian_map_NEG)
+{
+  std::unordered_map<const luci::CircleNode *, std::vector<float>> hessian_map;
+  auto ctx = std::make_unique<luci::QuantizeWeightsWithGPTQPass::Context>();
+  {
+    ctx->input_model_dtype = loco::DataType::FLOAT32;
+    ctx->output_model_dtype = loco::DataType::U8;
+    ctx->granularity = luci::QuantizationGranularity::ChannelWise;
+  }
+
+  luci::QuantizeWeightsWithGPTQPass pass(std::move(ctx), &hessian_map);
+  // Expect no exception, pass should handle empty hessian map gracefully
+  EXPECT_NO_THROW(pass.run(&_g));
+}
+
+// Negative test: Weights are not of type FLOAT32
+TEST_F(QuantizeWeightsWithGPTQPassTest, run_with_non_float_weights_NEG)
+{
+  // Change the weights to non-float32
+  luci::CircleConst *weight = nullptr;
+  for (auto node : loco::all_nodes(&_g))
+  {
+    if (auto const_node = dynamic_cast<luci::CircleConst *>(node))
+    {
+      if (const_node->name() == "nconv/filter")
+      {
+        weight = const_node;
+        break;
+      }
+    }
+  }
+  ASSERT_NE(weight, nullptr);
+  // Set dtype to INT32
+  weight->dtype(loco::DataType::S32);
+
+  luci::QuantizeWeightsWithGPTQPass pass(loco::DataType::FLOAT32, loco::DataType::U8,
+                                         luci::QuantizationGranularity::ChannelWise);
+  // The pass should skip this node without exception
+  EXPECT_NO_THROW(pass.run(&_g));
+}
+
+// Positive test: Run pass with valid hessian map
+TEST_F(QuantizeWeightsWithGPTQPassTest, run_with_valid_hessian)
+{
+  // Create a hessian map with valid data
+  std::unordered_map<const luci::CircleNode *, std::vector<float>> hessian_map;
+  // Find the conv node
+  luci::CircleConv2D *conv = nullptr;
+  for (auto node : loco::all_nodes(&_g))
+  {
+    if (auto conv_node = dynamic_cast<luci::CircleConv2D *>(node))
+    {
+      conv = conv_node;
+      break;
+    }
+  }
+  ASSERT_NE(conv, nullptr);
+  const auto node_filter = loco::must_cast<luci::CircleConst *>(
+    loco::must_cast<const luci::CircleConv2D *>(conv)->filter());
+  // Create a dummy hessian vector
+  size_t weight_size = node_filter->size<loco::DataType::FLOAT32>();
+  std::vector<float> hessian(weight_size * weight_size, 0.0f);
+  for (size_t i = 0; i < weight_size; ++i)
+  {
+    hessian[i * weight_size + i] = 1.0f; // Identity matrix
+  }
+
+  hessian_map[conv] = hessian;
+
+  auto ctx = std::make_unique<luci::QuantizeWeightsWithGPTQPass::Context>();
+  {
+    ctx->input_model_dtype = loco::DataType::FLOAT32;
+    ctx->output_model_dtype = loco::DataType::U8;
+    ctx->granularity = luci::QuantizationGranularity::ChannelWise;
+  }
+
+  luci::QuantizeWeightsWithGPTQPass pass(std::move(ctx), &hessian_map);
+  EXPECT_NO_THROW(pass.run(&_g));
```

Comment:
Can you add some value tests? It is difficult for me to ensure that those implementations (cholesky things) are correct.